node/src/types/block.rs

```diff
@@ -923,14 +938,14 @@ impl FromBytes for Block {
 }
 
 impl BlockLike for Block {
-    fn deploys(&self) -> &Vec<DeployHash> {
-        self.deploy_hashes()
+    fn deploys(&self) -> Vec<&DeployHash> {
+        self.deploy_hashes().iter().collect()
```

Comment:
Just wondering – will `.collect()` allocate a new vector? Why not leave it as `&Vec`? Wouldn't it be a better in terms of performance and allocated memory?

Reply:
The reasoning behind this is that due to this PR, `deploys` and `transfers` become two separate collections in ProtoBlock, and we need to combine them for the `BlockLike` trait.

Reply:
We could still `iter()`, `chain()` and `collect()` at the call site instead. It sounds like even in the `BlockLike` impl we wouldn't want to allocate a new `Vec` for either collection alone.

Reply:
I chose to combine the two collections, `transfers` and `deploys` in the `BlockLike` trait's implementation, however calling into `chain` to combine them in the callsite would require that `BlockLike` also differentiate between `transfers()` and `deploys()`, meaning other implementers would also need to differentiate and `BlockLike` probably shouldn't impose that on implementers.

Reply:
OK (I must admit I don't have a good overview of where and how `BlockLike` is used.)